### PR TITLE
materialx: move to python3Packages

### LINKED ISSUE #325663: Build failure: OpenUSD
### Steps To Reproduce
1. Try to install the package `python3.11-openusd-24.05`
This is a dependency of some other package, but I don't know which one.

### Expected behavior
The package should install as usual.

### Additional context
```console
error:
       … while calling the 'head' builtin

         at /nix/store/2nnisw4pxbifisbkg58hrnis9ycs5ah1-source/lib/attrsets.nix:1575:11:

         1574|         || pred here (elemAt values 1) (head values) then
         1575|           head values
             |           ^
         1576|         else

       … while evaluating the attribute 'value'

         at /nix/store/2nnisw4pxbifisbkg58hrnis9ycs5ah1-source/lib/modules.nix:809:9:

          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          810|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Python version mismatch in 'python3.11-openusd-24.05':

       The Python derivation 'python3.11-openusd-24.05' depends on a Python derivation
       named 'python3.12-materialx-1.38.10', but the two derivations use different versions
       of Python:

               'python3.11-openusd-24.05' uses /nix/store/hyppxdhgqx7nbg394j97yhmi6vizl074-python3-3.11.9
           'python3.12-materialx-1.38.10' uses /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4

       Possible solutions:

         * If 'python3.12-materialx-1.38.10' is a Python library, change the reference to 'python3.12-materialx-1.38.10'
           in the buildInputs of 'python3.11-openusd-24.05' to use a 'python3.12-materialx-1.38.10' built from the same
           version of Python

         * If 'python3.12-materialx-1.38.10' is used as a tool during the build, move the reference to
           'python3.12-materialx-1.38.10' in 'python3.11-openusd-24.05' from buildInputs to nativeBuildInputs

         * If 'python3.12-materialx-1.38.10' provides executables that are called at run time, pass its
           bin path to makeWrapperArgs:

               makeWrapperArgs = [ "--prefix PATH : ${lib.makeBinPath [ materialx ] }" ];

        at /nix/store/yzkx7gx2cn73dh3c7zg5rvgkp3yli1cd-source/pkgs/development/python-modules/openusd/default.nix:53:3
┏━ 1 Errors: 
┃ error:
┃        … while calling the 'head' builtin
┃ 
┃          at /nix/store/2nnisw4pxbifisbkg58hrnis9ycs5ah1-source/lib/attrsets.nix:1575:11:
┃ 
┃          1574|         || pred here (elemAt values 1) (head values) then
┃          1575|           head values
┃              |           ^
┃          1576|         else
┃ 
┃        … while evaluating the attribute 'value'
┃ 
┃          at /nix/store/2nnisw4pxbifisbkg58hrnis9ycs5ah1-source/lib/modules.nix:809:9:
┃ 
┃           808|     in warnDeprecation opt //
┃           809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
┃              |         ^
┃           810|         inherit (res.defsFinal') highestPrio;
┃ 
┃        (stack trace truncated; use '--show-trace' to show the full trace)
┃ 
┃        error: Python version mismatch in 'python3.11-openusd-24.05':
┃ 
┃        The Python derivation 'python3.11-openusd-24.05' depends on a Python derivation
┃        named 'python3.12-materialx-1.38.10', but the two derivations use different versions
┃        of Python:
┃ 
┃                'python3.11-openusd-24.05' uses /nix/store/hyppxdhgqx7nbg394j97yhmi6vizl074-python3-3.11.9
┃            'python3.12-materialx-1.38.10' uses /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4
┃ 
┃        Possible solutions:
┃ 
┃          * If 'python3.12-materialx-1.38.10' is a Python library, change the reference to 'python3.12-materialx-1.38.10'
┃            in the buildInputs of 'python3.11-openusd-24.05' to use a 'python3.12-materialx-1.38.10' built from the same
┃            version of Python
┃ 
┃          * If 'python3.12-materialx-1.38.10' is used as a tool during the build, move the reference to
┃            'python3.12-materialx-1.38.10' in 'python3.11-openusd-24.05' from buildInputs to nativeBuildInputs
┃ 
┃          * If 'python3.12-materialx-1.38.10' provides executables that are called at run time, pass its
┃            bin path to makeWrapperArgs:
┃ 
┃                makeWrapperArgs = [ "--prefix PATH : ${lib.makeBinPath [ materialx ] }" ];
┃ 
┃         at /nix/store/yzkx7gx2cn73dh3c7zg5rvgkp3yli1cd-source/pkgs/development/python-modules/openusd/default.nix:53:3
┣━━━                                                             
┗━ ∑ ⚠ Exited with 1 errors reported by nix at 19:15:24 after 20s
```

### Notify maintainers
@ShaddyDC  

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.7-xanmod1, NixOS, 24.05 (Uakari), 24.05.20240706.49ee0e9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - nixpkgs: `/nix/store/as2lprmykdsqs41yijzdb19asn6iv4g9-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
fixes https://discourse.nixos.org/t/cant-rebuild-my-system-since-last-flake-update-thanks-to-python/48789
cc @radek-stasta

fixes https://github.com/NixOS/nixpkgs/issues/325663
cc @cig0 @Anifyuli @returnofblank

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dotlambda** [comment]
Actually it looks like we have to move materialx to python3Packages to fix this?
```
$ nix-build -A python311.pkgs.openusd
...
-- Found Boost: /nix/store/w4dfa1akyciynv6i9l0963fr386wbayk-boost-1.81.0-dev/lib/cmake/Boost-1.81.0/BoostConfig.cmake (found version "1.81.0")
-- Disabling boost-provided cmake config
-- Found Python3: /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python3.12 (found version "3.12.4") found components: Interpreter Development Development.Module Development.Embed
CMake Error at /nix/store/ll2b6cslly4v30im7qk34f9y21ziiz1i-cmake-3.29.6/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: python312) (found version "1.81.0")
```

**Comment by dotlambda** [comment]
> Does materialx have a Python API?

It does: https://github.com/AcademySoftwareFoundation/MaterialX/tree/main/python/Scripts

**Comment by gador** [comment]
It looks like this PR is based on an older version of `master` Please rebase (so the changes to blender are what they are on master now, see my PR #325292)

File: pkgs/applications/misc/blender/default.nix [hunk 1]
```diff
@@ -72,7 +72,7 @@
   pkg-config,
   potrace,
   pugixml,
-  python3Packages, # must use instead of python3.pkgs, see https://github.com/NixOS/nixpkgs/issues/211340
```
Comment (gador) [inline pkgs/applications/misc/blender/default.nix:75]: This needs to be reverted

File: pkgs/applications/misc/blender/default.nix [hunk 2]
```diff
@@ -87,6 +87,7 @@
 }:
 
 let
+  python3Packages = python311Packages;
```
Comment (gador) [inline pkgs/applications/misc/blender/default.nix:90]: This, too

**Review by gador — COMMENTED** [review]
The changes to `materialx` look good

File: pkgs/applications/misc/blender/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/applications/misc/blender/default.nix]: I removed the change that makes Blender use Python 3.11. You can discuss that with @amarshall in https://github.com/NixOS/nixpkgs/pull/326044.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]
just dropped materialx from inputs

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 326041` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>blender-hip: log was empty</li>
  </ul>
</details>
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>blender</li>
<li>materialx (python312Packages.materialx)</li>
<li>openusd</li>
<li>python311Packages.materialx</li>
<li>python311Packages.openusd</li>
<li>python312Packages.openusd</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
blender-hip:
<details>


warning: unused-argument
Unused argument: `darwin`.
Near pkgs/applications/misc/blender/default.nix:19:3:

```
   |
19 |   darwin,
   |   ^
```
</details>
blender:
<details>

got build log for '/nix/store/p8kii2a4sbzxl9cy26rv4prvyviqplm2-blender-4.1.1' from 'daemon'
When evaluating attribute ‘blender’:
warning: unused-argument
Unused argument: `darwin`.
Near pkgs/applications/misc/blender/default.nix:19:3:

```
   |
19 |   darwin,
   |   ^
```
</details>
</details>

**Pull request merged**